eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/UnmodifiableBiMap.java

```diff
@@ -0,0 +1,1119 @@
+        }
+        this.delegate = delegate;
+    }
+
+    public static <K, V> UnmodifiableBiMap<K, V> of(MutableBiMap<K, V> biMap)
+    {
+        return new UnmodifiableBiMap<>(biMap);
+    }
+
+    public static <K, V> UnmodifiableBiMap<K, V> of(Map<K, V> map)
+    {
+        if (map == null)
+        {
+            throw new IllegalArgumentException("cannot create a UnmodifiableBiMap for null");
+        }
+        return new UnmodifiableBiMap<>(new HashBiMap<>(map));
+    }
+
+    @Override
+    public MutableBiMap<K, V> newEmpty()
+    {
+        return this.delegate.newEmpty();
+    }
+
+    @Override
+    public MutableBiMap<V, K> inverse()
+    {
+        return this.delegate.inverse().asUnmodifiable();
+    }
+
+    @Override
+    public MutableBiMap<V, K> flipUniqueValues()
+    {
+        return this.delegate.flipUniqueValues();
+    }
+
+    @Override
+    public V getIfAbsent(K key, Function0<? extends V> function)
+    {
+        return this.delegate.getIfAbsent(key, function);
+    }
+
+    @Override
+    public V getIfAbsentValue(K key, V value)
+    {
+        return this.delegate.getIfAbsentValue(key, value);
+    }
+
+    @Override
+    public <P> V getIfAbsentWith(K key, Function<? super P, ? extends V> function, P parameter)
+    {
+        return this.delegate.getIfAbsentWith(key, function, parameter);
+    }
+
+    @Override
+    public <A> A ifPresentApply(K key, Function<? super V, ? extends A> function)
+    {
+        return this.delegate.ifPresentApply(key, function);
+    }
+
+    @Override
+    public RichIterable<K> keysView()
+    {
+        return this.delegate.keysView();
+    }
+
+    @Override
+    public RichIterable<V> valuesView()
+    {
+        return this.delegate.valuesView();
+    }
+
+    @Override
+    public RichIterable<Pair<K, V>> keyValuesView()
+    {
+        return this.delegate.keyValuesView();
+    }
+
+    @Override
+    public ImmutableBiMap<K, V> toImmutable()
+    {
+        return this.delegate.toImmutable();
+    }
+
+    @Override
+    public MutableSetMultimap<V, K> flip()
+    {
+        return this.delegate.flip();
+    }
+
+    @Override
+    public V put(K key, V value)
+    {
+        throw new UnsupportedOperationException("Cannot call put() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public V remove(Object key)
+    {
+        throw new UnsupportedOperationException("Cannot call remove() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> m)
+    {
+        throw new UnsupportedOperationException("Cannot call putAll() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public void clear()
+    {
+        throw new UnsupportedOperationException("Cannot call clear() on " + this.getClass().getSimpleName());
+    }
+
+    @Override
+    public Set<K> keySet()
+    {
+        return Collections.unmodifiableSet(this.delegate.keySet());
+    }
```

Comment:
In terms of ordering, I think it would be best to keep all the views together (keySet near inverse for example), and all the methods which throw Unsupported together.